src/basic/chase-symlinks.c

```diff
@@ -689,3 +689,36 @@ int chase_symlinks_and_fopen_unlocked(
 
         return 0;
 }
+
+int chase_symlinks_and_unlink(
+                const char *path,
+                const char *root,
+                ChaseSymlinksFlags chase_flags,
+                int unlink_flags,
+                char **ret_path) {
+
+        _cleanup_free_ char *p = NULL, *dir = NULL, *fname = NULL;
+        _cleanup_close_ int fd = -1;
+        int r;
+
+        assert(path);
+
+        r = path_extract_directory(path, &dir);
+        if (r < 0)
+                return r;
+        r = path_extract_filename(path, &fname);
+        if (r < 0)
+                return r;
+
+        fd = chase_symlinks_and_open(dir, root, chase_flags, O_DIRECTORY|O_CLOEXEC, ret_path ? &p : NULL);
```

Comment:
after thinking about this a pit, i think we should use `O_PATH|O_DIRECTORY|O_CLOEXEC` here.

why? because O_PATH fds can be created for inodes that we cannot open (i.e. where the access mode prohibits this). For dirs this means if we specify O_PATH we can get a ref to the dir even if we cannot enumerate it. and that's great, because it means that for dirs where we have the "x" bit, but no "r" bit, then we can still do the unlinkat() call.

hence, please add O_PATH to the flags here.